src/cargo/sources/git/oxide.rs

```diff
@@ -0,0 +1,315 @@
+pub fn open_repo(
+    repo_path: &std::path::Path,
+    config_overrides: Vec<BString>,
+    purpose: OpenMode,
+) -> Result<gix::Repository, gix::open::Error> {
+    gix::open_opts(repo_path, {
+        let mut opts = gix::open::Options::default();
+        opts.permissions.config = gix::permissions::Config::all();
+        opts.permissions.config.git_binary = purpose.needs_git_binary_config();
+        opts.with(gix::sec::Trust::Full)
+            .config_overrides(config_overrides)
+    })
+}
+
+/// Convert `git` related cargo configuration into the respective `git` configuration which can be
+/// used when opening new repositories.
+pub fn cargo_config_to_gitoxide_overrides(config: &Config) -> CargoResult<Vec<BString>> {
+    use gix::config::tree::{gitoxide, Core, Http, Key};
+    let timeout = HttpTimeout::new(config)?;
+    let http = config.http_config()?;
+
+    let mut values = vec![
+        gitoxide::Http::CONNECT_TIMEOUT.validated_assignment_fmt(&timeout.dur.as_millis())?,
+        Http::LOW_SPEED_LIMIT.validated_assignment_fmt(&timeout.low_speed_limit)?,
+        Http::LOW_SPEED_TIME.validated_assignment_fmt(&timeout.dur.as_secs())?,
+        // Assure we are not depending on committer information when updating refs after cloning.
+        Core::LOG_ALL_REF_UPDATES.validated_assignment_fmt(&false)?,
+    ];
+    if let Some(proxy) = &http.proxy {
+        values.push(Http::PROXY.validated_assignment_fmt(proxy)?);
+    }
+    if let Some(check_revoke) = http.check_revoke {
+        values.push(Http::SCHANNEL_CHECK_REVOKE.validated_assignment_fmt(&check_revoke)?);
+    }
+    if let Some(cainfo) = &http.cainfo {
+        values.push(
+            Http::SSL_CA_INFO.validated_assignment_fmt(&cainfo.resolve_path(config).display())?,
+        );
+    }
+    if let Some(user_agent) = &http.user_agent {
+        values.push(Http::USER_AGENT.validated_assignment_fmt(user_agent)?);
+    }
+    if let Some(ssl_version) = &http.ssl_version {
+        use crate::util::config::SslVersionConfig;
+        match ssl_version {
+            SslVersionConfig::Single(version) => {
+                values.push(Http::SSL_VERSION.validated_assignment_fmt(&version)?);
+            }
+            SslVersionConfig::Range(range) => {
+                values.push(
+                    gitoxide::Http::SSL_VERSION_MIN
+                        .validated_assignment_fmt(&range.min.as_deref().unwrap_or("default"))?,
+                );
+                values.push(
+                    gitoxide::Http::SSL_VERSION_MAX
+                        .validated_assignment_fmt(&range.max.as_deref().unwrap_or("default"))?,
+                );
+            }
+        }
+    }
+    if let Some(debug) = http.debug {
+        values.push(gitoxide::Http::VERBOSE.validated_assignment_fmt(&debug)?);
+    }
+    if let Some(multiplexing) = http.multiplexing {
+        let http_version = multiplexing.then(|| "HTTP/2").unwrap_or("HTTP/1.1");
```

Comment:
Found [these lines](https://github.com/rust-lang/cargo/blob/39c13e67a5962466cc7253d41bc1099bbcb224c3/src/cargo/core/package.rs#L738-L754) pointing out that old macOS is probably going to be broken. However, sparse registry [doesn't respect that](https://github.com/rust-lang/cargo/blob/39c13e67a5962466cc7253d41bc1099bbcb224c3/src/cargo/sources/registry/http_remote.rs#L554-L559). I'll revisit this inconsistency afterward.

Reply:
Poor old macOS. A user just reported the issue #11726 from sparse registry 🥲.

Reply:
A great catch, and what a coincidence that we are immediately reminded that the `try_old_curl!()` check is indeed required.

I have fixed this by making the setting of this option non-fatal (see [`gix-transport` release](https://github.com/Byron/gitoxide/releases/tag/gix-transport-v0.25.5)) while also nothing here why `try_old_curl!()` isn't required or feasible in this case: d885c17883815c8f6fb425231ac7e6a16c1439e0 .